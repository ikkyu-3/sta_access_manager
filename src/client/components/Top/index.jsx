// @flow
import type { TopProps } from "@types";

import * as React from "react";
import styles from "./style.css";

class Top extends React.Component<TopProps> {
  componentDidMount() {
    this.props.refresh();
    this.props.selectedMenu();
  }

  render() {
    const { top, label } = styles;
    return (
      <div className={top}>
        <div>カードをスキャンして下さい</div>
        <div className={label}>Created by hr-js</div>
      </div>
    );
  }
}
export default Top;
