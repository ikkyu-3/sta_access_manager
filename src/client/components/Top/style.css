@charset "utf-8";
.top {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 36px;
  font-weight: bold;
}

.label {
  position: absolute;
  right: 10px;
  bottom: 10px;
  font-size: 20px;
  font-weight: normal;
}
