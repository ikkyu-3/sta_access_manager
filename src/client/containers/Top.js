// @flow
import type { Dispatch } from "@types";

import { connect } from "react-redux";
import {
  selectTop,
  removeId,
  clearFirstName,
  clearLastName,
  clearMailAddress,
  hideInputValidationResult
} from "@actions";
import Top from "@components/Top";

const mapDispatchToProps = (dispatch: Dispatch) => ({
  refresh: () => {
    dispatch(removeId());
    dispatch(clearFirstName());
    dispatch(clearLastName());
    dispatch(clearMailAddress());
    dispatch(hideInputValidationResult());
  },
  selectedMenu: () => {
    dispatch(selectTop());
  }
});

export default connect(null, mapDispatchToProps)(Top);
