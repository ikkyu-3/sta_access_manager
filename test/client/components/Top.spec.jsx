import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Top from "@components/Top";

const fn = () => {};

describe("Top.jsxのテスト", () => {
  describe("スナップショット", () => {
    it("正しいレンダリング", () => {
      const tree = renderer.create(<Top refresh={fn} selectedMenu={fn} />);
      expect(tree).toMatchSnapshot();
    });
  });

  describe("コンポーネントのテスト", () => {
    const refreshMock = jest.fn();
    const selectedMenuMock = jest.fn();

    beforeAll(() =>
      shallow(<Top refresh={refreshMock} selectedMenu={selectedMenuMock} />));

    it("componentDidMountは、refreshを実行する", () => {
      expect(refreshMock).toBeCalled();
    });

    it("componentDidMountは、selectedMenuを実行する", () => {
      expect(selectedMenuMock).toBeCalled();
    });
  });
});
