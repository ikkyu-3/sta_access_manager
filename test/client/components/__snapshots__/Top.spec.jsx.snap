// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top.jsxのテスト スナップショット 正しいレンダリング 1`] = `
<div
  className="top"
>
  <div>
    カードをスキャンして下さい
  </div>
  <div
    className="label"
  >
    Created by hr-js
  </div>
</div>
`;
